src/handler.ts

```diff
@@ -40,84 +42,132 @@ export function getParamsFromUriPath(textDocumentURI: string): Params {
  * @param projects Sentry extension projects configurations.
  * @return Sentry projectID this document reports to.
  */
-export function matchSentryProject(params: Params, projects: SentryProject[]): SentryProject | undefined {
+export function matchSentryProject(params: Params, projects: SentryProject[]): Matched {
+    let matched: Matched = { project: undefined, fileMatched: undefined }
+
     if (!projects || !params.repo || !params.file) {
-        return undefined
+        return matched
     }
 
-    // Check if a Sentry project is associated with this document's repo and retrieve the project.
+    // Check if a Sentry project is associated with this document's repository and/or file and retrieve the project.
     // TODO: Handle the null case instead of using a non-null assertion !
     // TODO: Handle cases where the wrong project is matched due to similar repo name,
     // e.g. `sourcegraph-jetbrains` repo will match the `sourcegraph` project
+    for (const project of projects) {
```

Comment:
Yes, that is why I'll have to guide the user to write a more specific file matching pattern to avoid those kinds of mismatches. In that case, the setup would be e.g.: 
```
filters: { 
repositories: "/soucegraph\/sourcegraph"
file: "/share\/[add differentiating folder here]\/.*\[add differentiating file name here].[add differentiating file format here]"
```
As long as the logs aren't sent from the *same file* to two different Sentry projects, this would work out.

Reply:
@vanesa I think @lguychard's point was that they _are_ sent from the _same file_ to potentially two different Sentry projects.

The only way I see to solve this would be adding one decoration for each project, i.e. for a file in shared/ it would show "View sentry events (browser_extension) | View Sentry events (web)" because two projects (web and browser_extension) were configured.

Reply:
Hm, if that really is possible then that would complicate things a lot. What if a file reports to 3 or 4 different Sentry projects? 🤔

Reply:
It's definitely possible, with the `/shared` example from our own code this would happen quite a lot. Another example would be a library shared between several components/microservices.

Reply:
We'll address this in a future PR. Here is the issue to keep track of it: https://github.com/sourcegraph/sourcegraph-sentry/issues/35